src/libstd/sys/windows/thread.rs

```diff
@@ -92,14 +83,29 @@ pub unsafe fn yield_now() {
     SwitchToThread();
 }
 
+pub fn sleep(dur: Duration) {
+    unsafe {
+        if dur < Duration::zero() {
+            return yield_now()
+        }
+        let ms = dur.num_milliseconds();
+        // if we have a fractional number of milliseconds then add an extra
+        // millisecond to sleep for
+        let extra = dur - Duration::milliseconds(ms);
+        let ms = ms + if extra.is_zero() {0} else {1};
+        Sleep(ms as DWORD);
```

Comment:
Ah I knew I saw something a long time ago with more-than-millisecond precision, thanks for the pointer! I'd have to run some benchmarks though to see what the overhead of calling `CreateWaitableTimer` is with respect to calling `Sleep`. I'm not sure if it's desirable to create a separate timer for each call to `sleep`.